mailpoet/lib/Newsletter/Preview/SendPreviewController.php

```diff
@@ -103,4 +113,119 @@ public function sendPreview(NewsletterEntity $newsletter, string $emailAddress)
       throw new SendPreviewException($error);
     }
   }
+
+  /**
+   * Add sample WooCommerce order/customer data to context for preview.
+   * Uses WooCommerce's dummy data for consistent, predictable previews.
+   *
+   * @param array $context Existing context
+   * @return array Context with sample data added
+   */
+  private function addSampleDataToContext(array $context): array {
+    $dummyOrder = $this->getWooCommerceDummyOrder();
+    if ($dummyOrder instanceof \WC_Order) {
+      $context['order'] = $dummyOrder;
+    }
+
+    $dummyCustomer = $this->getWooCommerceDummyCustomer();
+    if ($dummyCustomer instanceof \WC_Customer) {
+      $context['customer'] = $dummyCustomer;
+    }
+
+    return $context;
+  }
+
+  /**
+   * Get a dummy WooCommerce order using WooCommerce's EmailPreview class.
+   * This reuses WooCommerce's placeholder generation logic.
+   *
+   * @return \WC_Order|null Dummy order or null if unavailable
+   */
+  private function getWooCommerceDummyOrder(): ?\WC_Order {
+    // Check if WooCommerce EmailPreview class exists
+    if (!class_exists('Automattic\WooCommerce\Internal\Admin\EmailPreview\EmailPreview')) {
+      return null;
+    }
+
+    try {
+      // Access WooCommerce EmailPreview singleton
+      $emailPreview = \Automattic\WooCommerce\Internal\Admin\EmailPreview\EmailPreview::instance();
+
+      // Use reflection to call the private get_dummy_order() method
+      $reflectionClass = new \ReflectionClass($emailPreview);
+      $method = $reflectionClass->getMethod('get_dummy_order');
+      $method->setAccessible(true);
+
+      $dummyOrder = $method->invoke($emailPreview);
+
+      return $dummyOrder instanceof \WC_Order ? $dummyOrder : null;
+    } catch (\Throwable $e) {
+      // If reflection fails, try the filter as fallback
+      $dummyOrder = $this->wp->applyFilters('woocommerce_email_preview_dummy_order', null, null);
```

Comment:
What's the expectation here? Applying the same filter as WooCommerce core applies, with `null` will just return `null` no?